### PR TITLE
convert stateful class component to function component

### PR DESCRIPTION
**Pull request by nellbhealy**
Used hooks to convert stateful class component to function component
@averyvery @solomonhawk

### CODE REVIEW

File: src/components/board.js [hunk 1]
```diff
@@ -1,22 +1,26 @@
...
+const Board = props => (
+    <div className="root">
+        <div className="grid-container">{props.lights}</div>
+        <p>{props.numMoves}</p>
+        <button data-testid="reset" onClick={props.reset}>
+            Reset
+        </button>
+        {props.hasWon && (
+            <div data-testid="win-div">
+                Congratulations, you Won! Click reset to play again.
```
Comment (averyvery) [inline src/components/board.js]: Curious about @solomonhawk's thoughts on this — usually, if I have a long string like this and need to test for it, I end up setting/exporting the string as a const like `WIN_TEXT = 'Congratulations....'`. That way, there's no duplication and changing the text in `board.js` won't cause the test to break.

File: src/containers/board-container.js [hunk 2]
```diff
@@ -1,42 +1,46 @@
...
     getNewBoard,
     flipTheLights,
     getLights,
     checkForWin,
 } from '../game-logic.js';
 
-export class BoardContainer extends React.Component {
-    state = {
+const BoardContainer = props => {
+    const [data, setData] = useState({
```
Comment (averyvery) [inline src/containers/board-container.js:12]: :+1:

**Review by averyvery — APPROVED** [review]
Looks good to me!

File: src/components/board.js [hunk 3]
```diff
@@ -1,22 +1,24 @@
...
+export const WIN_TEXT = 'Congratulations, you Won! Click reset to play again.';
+
+const Board = props => (
+    <div className="root">
+        <div className="grid-container">{props.lights}</div>
+        <p>{props.numMoves}</p>
+        <button data-testid="reset" onClick={props.reset}>
+            Reset
+        </button>
+        {props.hasWon && <div data-testid="win-div">{WIN_TEXT}</div>}
```
Comment (solomonhawk) [inline src/components/board.js:13]: Looks great - a note specifically about **React Native**:

This is a bit of foreshadowing... this pattern works great in React (web), but has a tendency to cause problems in React Native (specifically referring to `[expression] && [expression]` which of course we use to coerce some value into a boolean in order to conditionally evaluate and return the following expression). If the first expression is falsey, then the whole combined expression evaluates to `false` which React Native chokes on (can't render text outside of a `<Text />` component) - again totally fine in React web. For RN, instead you use a ternary and explicitly return `null`: 

```jsx
{props.hasWon ? <whatever /> : null}
```

**Review by solomonhawk — APPROVED** [review]
👍 LGTM

**Pull request merged**